EKrumynas/Services/ShoppingCartService.cs

```diff
@@ -0,0 +1,132 @@
+﻿using AutoWrapper.Wrappers;
+using EKrumynas.Data;
+using EKrumynas.Models;
+using EKrumynas.Services.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace EKrumynas.Services
+{
+    public class ShoppingCartService : IShoppingCartService
+    {
+        private readonly EKrumynasDbContext _context;
+
+        public ShoppingCartService(EKrumynasDbContext context)
+        {
+            _context = context;
+        }
+        public async Task<ShoppingCart> CreateCart(ShoppingCart cart)
+        {
+            _context.ShoppingCarts.Add(cart);
+            await _context.SaveChangesAsync();
+
+            return cart;
+        }
+
+        public async Task<ShoppingCart> DeleteById(int id)
+        {
+            ShoppingCart found = await _context.ShoppingCarts
+                .FirstOrDefaultAsync(x => x.Id == id);
+            if (found == null)
+            {
+                throw new ApiException(
+                    statusCode: 400,
+                    message: "Incorrect request data");
+            }
+
+            _context.Remove(found);
+            await _context.SaveChangesAsync();
+
+            return found;
+        }
+
+        public Task<ShoppingCart> DeleteItemById(int cartId, int id)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public async Task<IList<ShoppingCart>> GetAll()
+        {
+            return await _context.ShoppingCarts
+                .Include(p => p.Plants)
+                    .ThenInclude(plq => plq.Quantity)
+                .Include(p => p.Pots)
+                    .ThenInclude(pq => pq.Quantity)
+                .Include(b => b.Bouquets)
+                    .ThenInclude(bq => bq.Quantity)
+                .Include(s => s.Status)
+                .ToListAsync();
+        }
+
+        public async Task<ShoppingCart> GetCartById(int id)
+        {
+            return await _context.ShoppingCarts
+                .Include(p => p.Plants)
+                    .ThenInclude(plq => plq.Quantity)
+                .Include(p => p.Pots)
+                    .ThenInclude(pq => pq.Quantity)
+                .Include(p => p.Bouquets)
+                    .ThenInclude(bq => bq.Quantity)
+                .Include(s => s.Status)
+                .FirstOrDefaultAsync(p => p.Id == id);
+        }
+
+        public async Task<ShoppingCart> UpdateCart(int cartId, PotCartItem pot)
+        {
+            ShoppingCart cart = await _context.ShoppingCarts
+                .Include(p => p.Plants)
+                    .ThenInclude(plq => plq.Quantity)
```

Comment:
Do we need this .ThenInclude? I think it's unnecessary
Same for others .ThenIncludes.

Suggest to test it without it, I might be wrong though

Reply:
Take it into consideration after the merge. @GabijaBurok